src/jbi/bugzilla_objects.py

```diff
@@ -0,0 +1,247 @@
+        """Get count of whiteboard labels"""
+        if self.whiteboard is not None:
+            return self.whiteboard.count("[")
+        return 0
+
+    def get_whiteboard_as_list(self):
+        """Convert string whiteboard into list, splitting on ']' and removing '['."""
+        if self.whiteboard is not None:
+            split_list = self.whiteboard.replace("[", "").split("]")
+            return [x.strip() for x in split_list if x not in ["", " "]]
+        return []
+
+    def get_whiteboard_with_brackets_as_list(self):
+        """Convert string whiteboard into list, splitting on ']' and removing '['; then re-adding."""
+        wb_list = self.get_whiteboard_as_list()
+        if wb_list is not None and len(wb_list) > 0:
+            return [f"[{element}]" for element in wb_list]
+        return []
+
+    def get_jbi_labels(self):
+        """bugzilla is a required label for SYNC project and is being used to prevent gh syncing"""
+        return (
+            ["bugzilla"]
+            + self.get_whiteboard_as_list()
+            + self.get_whiteboard_with_brackets_as_list()
+        )
+
+    def get_potential_whiteboard_config_list(self):
+        """Get all possible whiteboard_tag configuration values"""
+        converted_list: List = []
+        for whiteboard in self.get_whiteboard_as_list():
+            converted_tag = self.convert_wb_to_tag(whiteboard=whiteboard)
+            if converted_tag not in [None, "", " "]:
+                converted_list.append(converted_tag)
+
+        if len(converted_list) == 0:
+            return None
+        return converted_list
+
+    @staticmethod
+    def convert_wb_to_tag(whiteboard):
+        """Extract tag from whiteboard label"""
+        _exists = whiteboard not in (" ", "")
+        if not _exists:
+            return None
+        return whiteboard.split(sep="-", maxsplit=1)[0].lower()
+
+    def get_jira_issue_dict(self):
+        """Extract bug info as jira issue dictionary"""
+        type_map: dict = {"enhancement": "Task", "task": "Task", "defect": "Bug"}
+        return {
+            "summary": self.summary,
+            "labels": self.get_jbi_labels(),
+            "issuetype": {"name": type_map.get(self.type, "Task")},
+        }
+
+    def extract_from_see_also(self):
+        """Extract Jira Issue Key from see_also if jira url present"""
+        if not self.see_also and len(self.see_also) > 0:
+            return None
+
+        for url in self.see_also:  # pylint: disable=not-an-iterable
+            try:
+                parsed_url: ParseResult = urlparse(url=url)
+                expected_hosts = ["jira", "atlassian"]
+
+                if any(  # pylint: disable=use-a-generator
+                    [part in expected_hosts for part in parsed_url.hostname.split(".")]
+                ):
+                    parsed_jira_key = parsed_url.path.split("/")[-1]
+                    return parsed_jira_key
+            except Exception:  # pylint: disable=broad-except
```

Comment:
Can't we list exceptions here?

Reply:
We could, I believe there's a possibility of an AttributeError; I'm not sure all the possibilities, but I would feel more comfortable leaving this as is _for now._

Reply:
We don't even want to log anything?